BL12-381 Integration

PlamenHristov opened the pull request:
### What

Added: 
- BLS12_G1ADD
- BLS12_G1MUL
- BLS12_G1MULTIEXP
- BLS12_G2ADD
- BLS12_G2MUL
- BLS12_G2MULTIEXP

Missing:

- BLS12_PAIRING
- BLS12_MAP_FP_TO_G1
- BLS12_MAP_FP2_TO_G2
- BLS12_HASH_TO_G1
- BLS12_HASH_TO_G2

### Why

Because of issue  #779

### Known limitations

- Missing tests
- Panics if points are invalid
- Implementation is likely to be imperfect

jayz22 commented:
Some general comments. I'm not too familiar with the implementation details yet.

None of these functions can use existing metering cost types, for each of these functions {p1, p2}_{add, mul, serialize}, a new cost types need to be added, calibrated and charged.
(I also remember @graydon mentioned something we need to be careful about when it comes to FFI, but I can't recall exactly)
These functions need to be exposed via the env, via env.json entry and a corresponding implementation in host.rs.
It would be nice to have some tests, not comprehensive coverage but for sanity checks, to make sure they are doing what's expected and there are no obvious issues.

PlamenHristov commented:
1. Implemented the missing:
- BLS12_PAIRING
- BLS12_MAP_FP_TO_G1
- BLS12_MAP_FP2_TO_G2
- BLS12_HASH_TO_G1
- BLS12_HASH_TO_G2

2.  Added tests 

Ready for review in terms of the implementation. 


Still requires adding budgeting of individual operations and charging them in [stellar-xdr](https://github.com/stellar/rs-stellar-xdr)
